frontend/src/metabase/public/components/EmbedModal/EmbedModal.tsx

```diff
@@ -0,0 +1,61 @@
+import { useState } from "react";
+import { t } from "ttag";
+import { useSelector } from "metabase/lib/redux";
+import { getApplicationName } from "metabase/selectors/whitelabel";
+import Modal from "metabase/components/Modal";
+import { Box } from "metabase/ui";
+import * as MetabaseAnalytics from "metabase/lib/analytics";
+import { ModalContentActionIcon } from "metabase/components/ModalContent";
+import { color } from "metabase/lib/colors";
+
+import type { EmbedType } from "./EmbedModal.types";
+import { EmbedModalHeader } from "./EmbedModal.styled";
+
+interface EmbedModalProps {
+  isOpen?: boolean;
+  onClose: () => void;
+  children: ({
+    embedType,
+    setEmbedType,
+  }: {
+    embedType: EmbedType;
+    setEmbedType: (type: EmbedType) => void;
+  }) => JSX.Element;
+}
+
+export const EmbedModal = ({ children, isOpen, onClose }: EmbedModalProps) => {
+  const [embedType, setEmbedType] = useState<EmbedType>(null);
+  const applicationName = useSelector(getApplicationName);
+
+  const isEmbeddingStage = embedType === "application";
```

Comment:
In the Milestone 4 PR I'm thinking of a better way to name these booleans because I want to make sure things aren't too confusing. I think putting `isEmbeddingStage` will be confusing as well since everything is named `Embed<something>` so it's not immediately obvious. Leave this as it is but if you have any ideas for how we can improve the naming here I'm all ears